### PR TITLE
[Merged by Bors] - fix: discard linarith wrong type equalities, style

### LINKED ISSUE #2610: `linarith` not ignoring irrelevant (in)equalities
see https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/linarith.20and.20ENNReal and the mwe of @eric-wieser there:

```lean
import Mathlib.Tactic.Linarith

example : 0 ≤ 1 := by linarith -- ok

example (s : Set ℕ) (h : s = ∅) : 0 ≤ 1 :=
by linarith
/-
failed to synthesize
  AddGroup (Set ℕ)
-/
```
Seems like `linarith` should ignore (in)equalities in the context it doesn't get.

### PR DESCRIPTION
**Pull request by digama0**
Mostly style fixes in linarith, but also fixes #2610

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linarith/Preprocessing.lean [hunk 1]
```diff
@@ -234,43 +232,42 @@ and turns it into a proof of a comparison `_ R 0`, where `R ∈ {=, ≤, <}`.
...
 `compWithZero h` takes a proof `h` of an equality, inequality, or negation thereof,
 and turns it into a proof of a comparison `_ R 0`, where `R ∈ {=, ≤, <}`.
  -/
-def compWithZero : Preprocessor :=
-{ name := "make comparisons with zero",
-  transform := fun e =>
-  return [← rearrangeComparison e] }
+def compWithZero : Preprocessor where
+  name := "make comparisons with zero"
+  transform e := try pure [← rearrangeComparison e] catch _ => pure []
```
Comment (digama0) [inline Mathlib/Tactic/Linarith/Preprocessing.lean:270]: this is the actual fix for #2610

Comment (eric-wieser) [reply Mathlib/Tactic/Linarith/Preprocessing.lean:270]: Isn't catching everything a bit risky? Shouldn't we only catch typeclass failures?

File: Mathlib/Tactic/Linarith/Preprocessing.lean [hunk 2]
```diff
@@ -414,6 +411,6 @@ so the size of the list may change.
 -/
 def preprocess (pps : List GlobalBranchingPreprocessor) (g : MVarId) (l : List Expr) :
     MetaM (List Branch) :=
-  pps.foldlM (fun ls pp => do pure (← ls.mapM $ fun b => do pp.process b.1 b.2).join) [(g, l)]
+  pps.foldlM (fun ls pp => return (← ls.mapM fun (g, l) => do pp.process g l).join) [(g, l)]
```
Comment (eric-wieser) [inline Mathlib/Tactic/Linarith/Preprocessing.lean:414]: This will conflict with #2605

File: Mathlib/Tactic/Linarith/Verification.lean [hunk 3]
```diff
@@ -213,9 +213,9 @@ def proveFalseByLinarith (cfg : LinarithConfig) : MVarId → List Expr → MetaM
     let certificate : Std.HashMap Nat Nat ← try
       oracle comps max_var
     catch e =>
-      trace[linarith] m!"{e.toMessageData}"
+      trace[linarith] e.toMessageData
       throwError "linarith failed to find a contradiction"
-    trace[linarith] m!"linarith has found a contradiction: {certificate.toList}"
```
Comment (eric-wieser) [inline Mathlib/Tactic/Linarith/Verification.lean:218]: Is the `m!` implied by `trace`?

File: Mathlib/Tactic/Linarith/Preprocessing.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Tactic/Linarith/Preprocessing.lean]: Okay, I've changed this to only catch synthesis exceptions, but it is probably clunky.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**